ldes-server-compaction/src/main/java/be/vlaanderen/informatievlaanderen/ldes/server/compaction/application/services/CompactedFragmentCreator.java

```diff
@@ -0,0 +1,86 @@
+package be.vlaanderen.informatievlaanderen.ldes.server.compaction.application.services;
+
+import be.vlaanderen.informatievlaanderen.ldes.server.compaction.domain.repository.ViewCollection;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.model.LdesFragmentIdentifier;
+import be.vlaanderen.informatievlaanderen.ldes.server.domain.model.TreeRelation;
+import be.vlaanderen.informatievlaanderen.ldes.server.fetching.entities.MemberAllocation;
+import be.vlaanderen.informatievlaanderen.ldes.server.fetching.repository.AllocationRepository;
+import be.vlaanderen.informatievlaanderen.ldes.server.fragmentation.FragmentationStrategyImpl;
+import be.vlaanderen.informatievlaanderen.ldes.server.fragmentation.entities.Fragment;
+import be.vlaanderen.informatievlaanderen.ldes.server.fragmentation.repository.FragmentRepository;
+import io.micrometer.observation.Observation;
+import io.micrometer.observation.ObservationRegistry;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import static be.vlaanderen.informatievlaanderen.ldes.server.domain.constants.RdfConstants.GENERIC_TREE_RELATION;
+
+@Service
+public class CompactedFragmentCreator {
+	private final FragmentRepository fragmentRepository;
+	private final FragmentationStrategyImpl fragmentationStrategy;
```

Comment:
Can we use the interface here instead of the impl?

Reply:
updated